core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageAutoFeatureStep.java

```diff
@@ -618,26 +656,22 @@ private MethodDescriptor createRegisterSerializationForClassMethod(ClassCreator
                 ofMethod(Constructor.class, "getDeclaringClass", Class.class),
                 newSerializationConstructor);
 
-        ResultHandle lookupMethod = tc.invokeStaticMethod(
-                ofMethod("com.oracle.svm.util.ReflectionUtil", "lookupMethod", Method.class, Class.class, String.class,
-                        Class[].class),
-                tc.loadClass(Constructor.class), tc.load("getConstructorAccessor"),
+        ResultHandle getConstructorAccessor = tc.invokeStaticMethod(
+                lookupMethod, tc.loadClass(Constructor.class), tc.load("getConstructorAccessor"),
                 tc.newArray(Class.class, tc.load(0)));
 
         ResultHandle accessor = tc.invokeVirtualMethod(
-                ofMethod(Method.class, "invoke", Object.class, Object.class,
-                        Object[].class),
-                lookupMethod, newSerializationConstructor,
+                invokeMethodDescriptor, getConstructorAccessor, newSerializationConstructor,
                 tc.newArray(Object.class, tc.load(0)));
 
         tc.invokeVirtualMethod(
                 ofMethod("com.oracle.svm.reflect.serialize.SerializationSupport", "addConstructorAccessor",
                         Object.class, Class.class, Class.class, Object.class),
                 serializationSupport, clazz, newSerializationConstructorClass, accessor);
-        tc.invokeStaticMethod(
-                ofMethod("com.oracle.svm.reflect.serialize.hosted.SerializationFeature", "addReflections", void.class,
-                        Class.class, Class.class),
-                clazz, objectClass);
+        ResultHandle addRedlectionsArgs2 = tc.newArray(Class.class, tc.load(2));
+        tc.writeArrayValue(addRedlectionsArgs2, 0, clazz);
+        tc.writeArrayValue(addRedlectionsArgs2, 1, objectClass);
+        tc.invokeVirtualMethod(invokeMethodDescriptor, addReflectionsLookupMethod, tc.loadNull(), addRedlectionsArgs2);
```

Comment:
What is the `tc.loadNull()` supposed to be?

Reply:
`invoke`'s first argument is the `this` object, so since this is a static method I am passing `null` as the first argument.

> If the underlying method is static, then the specified obj argument is ignored. It may be null. 
See https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/reflect/Method.html#invoke(java.lang.Object,java.lang.Object...)